common/unboundedSemaphore.go

```diff
@@ -24,17 +24,23 @@ import "sync"
 
 // UnboundedSemaphore operates similarly to traditional Go channel-based semaphores,
 // but without any limitation on the number of resources that can be released
+//
+// See also: https://godoc.org/github.com/dropbox/godropbox/sync2#NewUnboundedSemaphore
+//
 type UnboundedSemaphore struct {
-	resCount    int
-	once        sync.Once
-	c           *sync.Cond
-	m           sync.Mutex // Don't use this directly, use through s.c.L
-	acquireLock sync.Mutex // Used by acquirer for credit accumulation
+	resCount        int
+	once            sync.Once
+	c               *sync.Cond
+	m               sync.Mutex // need to name this so it's not exported. Don't use this directly, use through s.c.L
+	accumulatorLock sync.Mutex // Used by acquirer for credit accumulation
 }
 
 // Release releases n resources
 func (s *UnboundedSemaphore) Release(n int) {
 	s.once.Do(s.init)
```

Comment:
While this is interesting in that it looks similar to a "mutex" (in not needing a New), I think it is a little weird that you have to start by dong a "Release(n)" to being using the semaphore.

I personally think it would be good to have a "New(initialCount int)" .. that takes in an "initial" count for the semaphore.

Probably not a fair comparison, since these are different platforms, nevertheless, for reference these take in an "initial" count:
java semaphore: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Semaphore.html
win32 semaphore: https://msdn.microsoft.com/en-us/library/windows/desktop/ms682438(v=vs.85).aspx